### PR TITLE
Remove use of persistent token in CI

### PR DESCRIPTION
**Pull request by ocaisa**
Some temporary changes included here to actually test this

### CODE REVIEW

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: check, build, deploy hpc-carpentry.org
 
 on:
+  pull_request: []
```
Comment (ocaisa) [inline .github/workflows/build-and-deploy.yml]: Remove on CI success

File: .github/workflows/build-and-deploy.yml [hunk 2]
```diff
@@ -59,8 +64,7 @@ jobs:
...
 
       - name: Build and deploy gh-pages branch
-        uses: helaili/jekyll-action@2.0.5
-        env:
-          JEKYLL_PAT: ${{ secrets.JEKYLL_PAT }}
+        uses: helaili/jekyll-action@2.2.0
         with:
-          target_branch: 'gh-pages'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          target_branch: 'gh-pages_check'
```
Comment (ocaisa) [inline .github/workflows/build-and-deploy.yml]: Revert branch on CI success

**Comment by ocaisa** [comment]
Confirmed in https://github.com/hpc-carpentry/hpc-carpentry.github.io/runs/3684415263 (and the resulting branch https://github.com/hpc-carpentry/hpc-carpentry.github.io/tree/gh-pages_check) that this works as expected, reverting the changes to check this...

**Comment by ocaisa** [comment]
When this is merged we should also remove the `gh-pages_check` branch, and delete the secret that we use.

**Review by reid-a — APPROVED** [review]
Look reasonable to me.

**Comment by tkphd** [comment]
To clarify, `secrets.GITHUB_TOKEN` is generated at build-time, not stored; and removing the stored `secrets.JEKYLL_PAT` means we have no "state" "secrets" in the repo/settings to leak?

**Comment by ocaisa** [comment]
@tkphd Yes that's right, the `GITHUB_TOKEN` token is created with the Github Action and is only valid as long as the Action, see https://docs.github.com/en/actions/security-guides/automatic-token-authentication for the full documentation.

If the existing `secrets.JEKYLL_PAT` was leaked, that token would have full write permissions to the repo.

**Review by tkphd — APPROVED** [review]
Excellent work!

**Pull request merged**